cachi2/core/package_managers/pip.py

```diff
@@ -30,6 +30,7 @@
 )
 
 import tomli
```

Comment:
**Commit message:**

>  Removing pkg_resources and subclassing
directly from packaging.Requirement.

You left out the part where you actually subclass `packaging.Requirement` in your implementation ;)

> setuptools has been removed because it was only used by pkg_resources

Commits should be contained to a single logical change, so from that POV ^this removal should be placed in a separate commit. I'm surprised this worked and passed the tests since we depend on setuptools as the built backend, but given it's specified as `required` at the top of `pyproject.toml` it'll get installed anyway, so if the CI's green I'm okay with dropping it from the deps list like you do.

> Resolves #486

Please put the actual URL in the commit message rather than a numeric ID so that when browsing git history locally one can access the URL directly.

Reply:
> but given it's specified as required at the top of pyproject.toml it'll get installed anyway
Yes, it's this. That's why I recommended removing it from direct dependencies - it was *only* there for `pkg_resources`, we're removing `pkg_resources`, so 'setuptools' should be removed from direct dependencies *in the same commit*

Reply:
> we're removing pkg_resources, so 'setuptools' should be removed from direct dependencies in the same commit

Not necessarily in the same one, if you mix everything that looks like it's somewhat related in a single commit it just creates noise, you only ever absolutely need to pair changes which would otherwise lead to broken builds or failing CI (on that particular commit) within a single commit which this is not the case, it'll work just fine and so the dependency can be dropped later and make the changes more linear, gradual and cleaner to review.